sdg/inputs/InputMetaFiles.py

```diff
@@ -0,0 +1,136 @@
+                meta[language] = translated_meta
+
+
+    def add_git_dates(self, meta, filepath):
+        git_update = self.get_git_updates(filepath)
+        for k in git_update.keys():
+            meta[k] = git_update[k]
+
+
+    def get_git_updates(self, filepath):
+        meta_update = self.get_git_update(filepath)
+        updates = {
+            'national_metadata_update_url_text': meta_update['date'] + ': see changes on GitHub',
+            'national_metadata_update_url': meta_update['commit_url']
+        }
+
+        indicator_id = self.convert_path_to_indicator_id(filepath)
+        data_filename = self.git_data_filemask.replace('*', indicator_id)
+        src_dir = os.path.dirname(os.path.dirname(self.path_pattern))
+        data_filepath = os.path.join(src_dir, self.git_data_dir, data_filename)
+        if os.path.isfile(data_filepath):
+            data_update = self.get_git_update(data_filepath)
+            updates['national_data_update_url_text'] = data_update['date'] + ': see changes on GitHub'
+            updates['national_data_update_url'] = data_update['commit_url']
+        
+        return updates
+
+
+    def get_git_update(self, filepath):
+        """Change into the working directory of the file (it might be a submodule)
+        and get the latest git history"""
+        folder = os.path.split(filepath)[0]
+        
+        repo = git.Repo(folder, search_parent_directories=True)
+        # Need to translate relative to the repo root (this may be a submodule)
+        repo_dir = os.path.relpath(repo.working_dir, os.getcwd())
+        filepath = os.path.relpath(filepath, repo_dir)
+        commit = next(repo.iter_commits(paths=filepath, max_count=1))
+        git_date = str(commit.committed_datetime.date())
+        git_sha = commit.hexsha
+        # Turn the remote URL into a commit URL
+        remote = repo.remote().url
+        remote_bare = re.sub('^.*github\.com(:|\/)', '', remote).replace('.git','')
+        commit_url = 'https://github.com/'+remote_bare+'/commit/'+git_sha
+        
+        return {
+            'date': git_date,
+            'commit_url': commit_url
+        }
+    
+
+    def load_metadata_mapping(self):
+        mapping = None
+        if self.metadata_mapping is None:
+            mapping = {}
+        elif isinstance(self.metadata_mapping, dict):
+            mapping = self.metadata_mapping
+        # Otherwise assume it is a path to a file.
+        else:
+            extension = os.path.splitext(self.metadata_mapping)[1]
+            if extension.lower() == '.csv':
+                mapping = pd.read_csv(self.metadata_mapping, header=None, index_col=0, squeeze=True).to_dict()
+
+        if mapping is None:
+            raise Exception('Format of metadata_mapping should be a dict or a path to a CSV file.')
+
+        self.metadata_mapping = mapping
+
+
+    def apply_metadata_mapping(self, metadata):
+        for human_key in self.metadata_mapping:
```

Comment:
@brockfanning is this supposed to loop through rows or columns? I still seem to be having an issue with this